src/models/hooks/goal.js

```diff
@@ -331,10 +331,52 @@ const beforeUpdate = async (sequelize, instance, options) => {
   await preventCloseIfObjectivesOpen(sequelize, instance, options);
 };
 
+/**
+ * Creates a GoalStatusChange record for the initial status of a goal
+ * This ensures the creation event is always included in status history
+ */
+const createInitialStatusChange = async (sequelize, instance, options) => {
+  // get the creator collaborator type
+  const creatorType = await sequelize.models.CollaboratorType.findOne({
+    where: { name: GOAL_COLLABORATORS.CREATOR },
+    transaction: options.transaction,
+  });
+
+  // get the creator collaborator for this goal
+  const goalCollaborator = creatorType ? await sequelize.models.GoalCollaborator.findOne({
+    where: {
+      goalId: instance.id,
+      collaboratorTypeId: creatorType.id,
+    },
+    include: [
+      {
+        model: sequelize.models.User,
+        as: 'user',
+        attributes: ['id', 'name'],
+      },
+    ],
+    transaction: options.transaction,
+  }) : null;
+
+  // create a GoalStatusChange record for the initial status
+  await sequelize.models.GoalStatusChange.create({
+    goalId: instance.id,
+    userId: goalCollaborator?.userId || options.userId || null,
+    userName: goalCollaborator?.user?.name || null,
+    oldStatus: null,
+    newStatus: instance.status || 'Not Started',
+    reason: 'Goal created',
+    context: 'Creation',
+  }, {
+    transaction: options.transaction,
+  });
+};
```

Comment:
this had to be done because the design for goal history wants to see the original status event, so instead of faking it, it seemed better to actually create a status change entry